controllers/metal3machine_controller.go

```diff
@@ -182,43 +187,62 @@ func (r *Metal3MachineReconciler) reconcileNormal(ctx context.Context,
 	// If the Metal3Machine doesn't have finalizer, add it.
 	machineMgr.SetFinalizer()
 
-	// if the machine is already provisioned, return
+	// if the machine is already provisioned, update and return
 	if machineMgr.IsProvisioned() {
-		err := machineMgr.Update(ctx)
-		return ctrl.Result{}, err
+		errType := capierrors.UpdateMachineError
+		return checkMachineError(machineMgr, machineMgr.Update(ctx),
+			"Failed to update the Metal3Machine", errType,
+		)
 	}
 
+	errType := capierrors.CreateMachineError
```

Comment:
I would rather move the declaration close to where it is used, i.e. line# 205

Reply:
done, thanks